instruments/novatelGPS/ReceiverConfig/config_NAPxx45_DABxx78.txt
2015-12-29

8360045

GPSCARD "L12LV" "DAB08340078" "OEMV3G-4.00-X2T" "3.906" "3.000" "2013/Mar/14" "14:22:01"

COM1 115200 N 8 1 N OFF ON NOVATEL NOVATEL ON 
COM2 115200 N 8 1 N OFF ON NOVATEL NOVATEL ON
COM3 9600 N 8 1 N OFF ON

COM1 GPGGA ONTIME 0.100000 0.000000 NOHOLD
COM1 BESTPOSA ONTIME 0.100000 0.000000 NOHOLD
COM1 BESTVELA ONTIME 0.100000 0.000000 NOHOLD
COM1 GPRMC ONTIME 0.100000 0.000000 NOHOLD
COM1 TIMEA ONTIME 60.000000 0.000000 NOHOLD
COM2 RAWEPHEMB ONCHANGED 0.000000 0.000000 NOHOLD
COM2 RANGECMPB ONTIME 0.100000 0.000000 NOHOLD
COM2 BESTPOSB ONTIME 1.000000 0.000000 NOHOLD


LOG COM1 113 30.0 UNKNOWN 0 1.855 00000008 f702 9603
COM1 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 112 30.0 UNKNOWN 0 1.864 00000008 f702 9603
COM2 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 111 30.0 UNKNOWN 0 1.870 00000008 f702 9603
COM3 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 110 30.0 UNKNOWN 0 1.877 00000008 f702 9603
AUX RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 109 30.0 UNKNOWN 0 1.887 00000008 f702 9603
USB1 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 108 30.0 UNKNOWN 0 1.893 00000008 f702 9603
USB2 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

LOG COM1 107 30.0 UNKNOWN 0 1.901 00000008 f702 9603
USB3 RXSTATUSEVENTA ONNEW 0.000000 0.000000 HOLD

POSAVE COM1 106 30.0 UNKNOWN 0 1.525 00000008 f702 9603
OFF 0.00 0.0 0.0

ASSIGNALL COM1 105 30.0 UNKNOWN 0 1.908 00000008 f702 9603
GPSL5 IDLE 0 0 4500

RTKQUALITYLEVEL COM1 104 30.0 UNKNOWN 0 1.910 00000008 f702 9603
NORMAL

RTKVERIFIEDQUALITY COM1 103 30.0 UNKNOWN 0 1.916 00000008 f702 9603
NORMAL NORMAL

IONOCONDITION COM1 102 30.0 UNKNOWN 0 1.921 00000008 f702 9603
QUIET

SATCUTOFF COM1 101 30.0 UNKNOWN 0 1.923 00000008 f702 9603
ENABLE 20

SETRTCMRXVERSION COM1 100 30.0 UNKNOWN 0 1.927 00000008 f702 9603
V23

ADJUST1PPS COM1 99 30.0 UNKNOWN 0 1.933 00000008 f702 9603
OFF ONCE 0

CDGPSTIMEOUT COM1 98 30.0 UNKNOWN 0 1.937 00000008 f702 9603
AUTO 120.000000000 0.000000000

CLOCKADJUST COM1 97 30.0 UNKNOWN 0 1.939 00000008 f702 9603
ENABLE

CLOCKOFFSET COM1 96 30.0 UNKNOWN 0 1.942 00000008 f702 9603
0

CNOUPDATE COM1 95 30.0 UNKNOWN 0 1.946 00000008 f702 9603
DEFAULT

COM COM1 94 30.0 UNKNOWN 0 1.948 00000008 f702 9603
AUX 9600 N 8 1 N OFF ON

COM COM1 93 30.0 UNKNOWN 0 1.954 00000008 f702 9603
USB1 9600 N 8 1 N OFF ON

COM COM1 92 30.0 UNKNOWN 0 1.958 00000008 f702 9603
USB2 9600 N 8 1 N OFF ON

COM COM1 91 30.0 UNKNOWN 0 1.962 00000008 f702 9603
USB3 9600 N 8 1 N OFF ON

COMCONTROL COM1 90 30.0 UNKNOWN 0 1.966 00000008 f702 9603
COM1 RTS DEFAULT

COMCONTROL COM1 89 30.0 UNKNOWN 0 1.970 00000008 f702 9603
COM2 RTS DEFAULT

COMCONTROL COM1 88 30.0 UNKNOWN 0 1.974 00000008 f702 9603
COM3 RTS DEFAULT

CSMOOTH COM1 87 30.0 UNKNOWN 0 1.978 00000008 f702 9603
100 100

DATUM COM1 86 30.0 UNKNOWN 0 1.980 00000008 f702 9603
WGS84

LOCALIZEDCORRECTIONDATUM COM1 85 30.0 UNKNOWN 0 1.984 00000008 f702 9603
WGS84

DIFFCODEBIASCONTROL COM1 84 30.0 UNKNOWN 0 1.988 00000008 f702 9603
ENABLE

DGPSEPHEMDELAY COM1 83 30.0 UNKNOWN 0 1.990 00000008 f702 9603
120

DGPSTIMEOUT COM1 82 30.0 UNKNOWN 0 1.992 00000008 f702 9603
300

DGPSTXID COM1 81 30.0 UNKNOWN 0 1.996 00000008 f702 9603
AUTO "ANY"

DYNAMICS COM1 80 30.0 UNKNOWN 0 1.998 00000008 f702 9603
AIR

ECUTOFF COM1 79 30.0 UNKNOWN 0 2.016 00000008 f702 9603
5.0

GLOECUTOFF COM1 78 30.0 UNKNOWN 0 2.020 00000008 f702 9603
5.000000000

GLOCSMOOTH COM1 77 30.0 UNKNOWN 0 2.023 00000008 f702 9603
100 100

EXTERNALCLOCK COM1 76 30.0 UNKNOWN 0 2.027 00000008 f702 9603
DISABLE 5MHZ 0.000000000 0.000000000 0.000000000

FIX COM1 75 30.0 UNKNOWN 0 2.028 00000008 f702 9603
NONE -10000.00000000000 -10000.00000000000 -10000.0000

SETBESTPOSCRITERIA COM1 74 30.0 UNKNOWN 0 2.033 00000008 f702 9603
POS3D 0

FORCEGPSL2CODE COM1 73 30.0 UNKNOWN 0 2.037 00000008 f702 9603
DEFAULT

FREQUENCYOUT COM1 72 30.0 UNKNOWN 0 2.041 00000008 f702 9603
DISABLE 0 0

HDTOUTTHRESHOLD COM1 71 30.0 UNKNOWN 0 2.043 00000008 f702 9603
2.000000000

INTERFACEMODE COM1 70 30.0 UNKNOWN 0 2.047 00000008 f702 9603
COM1 NOVATEL NOVATEL ON

INTERFACEMODE COM1 69 30.0 UNKNOWN 0 2.049 00000008 f702 9603
COM2 NOVATEL NOVATEL ON

INTERFACEMODE COM1 68 30.0 UNKNOWN 0 2.055 00000008 f702 9603
AUX NOVATEL NOVATEL ON

INTERFACEMODE COM1 67 30.0 UNKNOWN 0 2.057 00000008 f702 9603
XCOM1 NOVATEL NOVATEL ON

INTERFACEMODE COM1 66 30.0 UNKNOWN 0 2.061 00000008 f702 9603
XCOM2 NOVATEL NOVATEL ON

INTERFACEMODE COM1 65 30.0 UNKNOWN 0 2.063 00000008 f702 9603
XCOM3 NOVATEL NOVATEL ON

INTERFACEMODE COM1 64 30.0 UNKNOWN 0 2.067 00000008 f702 9603
USB1 NOVATEL NOVATEL ON

INTERFACEMODE COM1 63 30.0 UNKNOWN 0 2.073 00000008 f702 9603
USB2 NOVATEL NOVATEL ON

INTERFACEMODE COM1 62 30.0 UNKNOWN 0 2.075 00000008 f702 9603
USB3 NOVATEL NOVATEL ON

MAGVAR COM1 61 30.0 UNKNOWN 0 2.079 00000008 f702 9603
CORRECTION 0.000000000 0.000000000

MARKCONTROL COM1 60 30.0 UNKNOWN 0 2.084 00000008 f702 9603
MARK1 ENABLE NEGATIVE 0 0

MARKCONTROL COM1 59 30.0 UNKNOWN 0 2.089 00000008 f702 9603
MARK2 ENABLE NEGATIVE 0 0

MOVINGBASESTATION COM1 58 30.0 UNKNOWN 0 2.091 00000008 f702 9603
DISABLE

NMEATALKER COM1 57 30.0 UNKNOWN 0 2.095 00000008 f702 9603
GP

PASSTOPASSMODE COM1 56 30.0 UNKNOWN 0 2.099 00000008 f702 9603
DISABLE OFF OFF DEFAULT 1.000000000

PDPFILTER COM1 55 30.0 UNKNOWN 0 2.101 00000008 f702 9603
DISABLE

POSTIMEOUT COM1 54 30.0 UNKNOWN 0 2.105 00000008 f702 9603
600

PPSCONTROL COM1 53 30.0 UNKNOWN 0 2.110 00000008 f702 9603
ENABLE NEGATIVE 1.000000000 1000

PSRSATLIMIT COM1 52 30.0 UNKNOWN 0 2.113 00000008 f702 9603
AUTO 0

PSRVELOCITYTYPE COM1 51 30.0 UNKNOWN 0 2.117 00000008 f702 9603
DOPPLER

REALTIME COM1 50 30.0 UNKNOWN 0 2.120 00000008 f702 9603
ON

RTKCOMMAND COM1 49 30.0 UNKNOWN 0 2.121 00000008 f702 9603
USE_DEFAULTS

RTKSOLUTION COM1 48 30.0 UNKNOWN 0 2.134 00000008 f702 9603
AUTO

RTKBASELINE COM1 47 30.0 UNKNOWN 0 2.136 00000008 f702 9603
UNKNOWN 0.0000000000000000 0.0000000000000000 0.0000000000000000 0.000000000

RTKELEVMASK COM1 46 30.0 UNKNOWN 0 2.138 00000008 f702 9603
AUTO 0.000000000

RTKNETWORK COM1 45 30.0 UNKNOWN 0 2.142 00000008 f702 9603
DISABLE 0

RTKDYNAMICS COM1 44 30.0 UNKNOWN 0 2.144 00000008 f702 9603
DYNAMIC

RTKIFTHRESH COM1 43 30.0 UNKNOWN 0 2.148 00000008 f702 9603
7000.000000000

RTKFIXRATE COM1 42 30.0 UNKNOWN 0 2.150 00000008 f702 9603
1.250000000

RTKL1BLTHRESHOLD COM1 41 30.0 UNKNOWN 0 2.152 00000008 f702 9603
3.000000000

RTKSVENTRIES COM1 40 30.0 UNKNOWN 0 2.154 00000008 f702 9603
24

RTKTIMEOUT COM1 39 30.0 UNKNOWN 0 2.156 00000008 f702 9603
60

SETIONOTYPE COM1 38 30.0 UNKNOWN 0 2.161 00000008 f702 9603
AUTO

SETNAV COM1 37 30.0 UNKNOWN 0 2.163 00000008 f702 9603
90.000000000 0.000000000 90.000000000 0.000000000 0.000000000 "FROM" "TO"

SETRTCMFKP COM1 36 30.0 UNKNOWN 0 2.167 00000008 f702 9603
V23

STATUSCONFIG COM1 35 30.0 UNKNOWN 0 2.170 00000008 f702 9603
PRIORITY STATUS 0

STATUSCONFIG COM1 34 30.0 UNKNOWN 0 2.173 00000008 f702 9603
PRIORITY AUX1 8

STATUSCONFIG COM1 33 30.0 UNKNOWN 0 2.176 00000008 f702 9603
PRIORITY AUX2 0

STATUSCONFIG COM1 32 30.0 UNKNOWN 0 2.179 00000008 f702 9603
SET STATUS 0

STATUSCONFIG COM1 31 30.0 UNKNOWN 0 2.182 00000008 f702 9603
SET AUX1 0

STATUSCONFIG COM1 30 30.0 UNKNOWN 0 2.185 00000008 f702 9603
SET AUX2 0

STATUSCONFIG COM1 29 30.0 UNKNOWN 0 2.188 00000008 f702 9603
CLEAR STATUS 0

STATUSCONFIG COM1 28 30.0 UNKNOWN 0 2.191 00000008 f702 9603
CLEAR AUX1 0

STATUSCONFIG COM1 27 30.0 UNKNOWN 0 2.194 00000008 f702 9603
CLEAR AUX2 0

TIMEWARP COM1 26 30.0 UNKNOWN 0 2.196 00000008 f702 9603
1.000000000

UNDULATION COM1 25 30.0 UNKNOWN 0 2.199 00000008 f702 9603
EGM96 0.000000000

USEREXPDATUM COM1 24 30.0 UNKNOWN 0 2.202 00000008 f702 9603
6378137.000 298.25722356280 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000

UTMZONE COM1 23 30.0 UNKNOWN 0 2.206 00000008 f702 9603
AUTO 0

SETBESTPOSTRANSITIONMODE COM1 22 30.0 UNKNOWN 0 2.208 00000008 f702 9603
DISABLE 1200

RTKANTENNA COM1 21 30.0 UNKNOWN 0 2.212 00000008 f702 9603
UNKNOWN DISABLE FALSE FALSE

USEHPPRECONVERGED COM1 20 30.0 UNKNOWN 0 2.216 00000008 f702 9603
DISABLE

HEIGHTMODEL COM1 19 30.0 UNKNOWN 0 2.218 00000008 f702 9603
DISABLE 0.000000000 1.000000000

WAASECUTOFF COM1 18 30.0 UNKNOWN 0 2.220 00000008 f702 9603
-5.000000000

WAASTIMEOUT COM1 17 30.0 UNKNOWN 0 2.224 00000008 f702 9603
AUTO 600.000000000 0.000000000

RTKSOURCE COM1 16 30.0 UNKNOWN 0 2.226 00000008 f702 9603
OMNISTAR ""

PSRDIFFSOURCE COM1 15 30.0 UNKNOWN 0 2.230 00000008 f702 9603
OMNISTAR ""

COM COM1 14 30.0 UNKNOWN 0 2.232 00000008 f702 9603
COM2 115200 N 8 1 N OFF ON

COM COM1 13 30.0 UNKNOWN 0 2.236 00000008 f702 9603
COM1 115200 N 8 1 N OFF ON

LOG COM1 12 30.0 UNKNOWN 0 2.243 00000008 f702 9603
COM2 RAWEPHEMB ONCHANGED 0.000000 0.000000 NOHOLD

LOG COM1 11 30.0 UNKNOWN 0 2.249 00000008 f702 9603
COM2 RANGECMPB ONTIME 0.100000 0.000000 NOHOLD

LOG COM1 10 30.0 UNKNOWN 0 2.256 00000008 f702 9603
COM2 BESTPOSB ONTIME 1.000000 0.000000 NOHOLD

LOG COM1 9 30.0 UNKNOWN 0 2.264 00000008 f702 9603
COM1 GPGGA ONTIME 0.100000 0.000000 NOHOLD

LOG COM1 8 30.0 UNKNOWN 0 2.273 00000008 f702 9603
COM1 BESTPOSA ONTIME 0.100000 0.000000 NOHOLD

LOG COM1 7 30.0 UNKNOWN 0 2.281 00000008 f702 9603
COM1 BESTVELA ONTIME 0.100000 0.000000 NOHOLD

LOG COM1 6 30.0 UNKNOWN 0 2.292 00000008 f702 9603
COM1 GPRMC ONTIME 0.100000 0.000000 NOHOLD

LOG COM1 5 30.0 UNKNOWN 0 2.298 00000008 f702 9603
COM1 TIMEA ONTIME 60.000000 0.000000 NOHOLD

ANTENNAPOWER COM1 4 30.0 UNKNOWN 0 2.308 00000008 f702 9603
OFF

COM COM1 3 30.0 UNKNOWN 0 2.310 00000008 f702 9603
COM3 9600 N 8 1 N OFF ON

INTERFACEMODE COM1 2 30.0 UNKNOWN 0 2.314 00000008 f702 9603
COM3 NOVATEL NOVATEL ON

SBASCONTROL COM1 1 30.0 UNKNOWN 0 2.318 00000008 f702 9603
ENABLE AUTO 0 NONE

ASSIGNLBAND COM1 0 30.0 UNKNOWN 0 2.324 00000008 f702 9603
OMNISTARAUTO 1545000000 1200

doc/HISTORY

Version number is changed in XmNimbus.
-------------------------------------------------------------------------
11/30/98	2.5.5
	MASP support.  Cell sizes are end-points now, not mid-points.
	260x changes/fixes, including removal physical bin-shift between
	ACCUM & CONC, now just adjust starting.

05/19/98	2.5.4
	netCDF mods, removed global defaults, and stored default with given
	variable.  Made nimbus check default directory for Defaults
	then local proj dir.  netCDF file version 1.1.  fixed fssp range
	affecting HRT sampled FRNG.  Cleaned up all processing of HRT
	sampled PMS1D (mostly housekeeping).  Corrections to 260x for
	HRT sampling also.  Modified to handle 30 minute time gaps before
	inserting new time period.  New AKRD calcoes for C130.

01/02/98	2.5.3
	netCDF mods.  PMSspecs recorded with AFSSP, etc.  Only Dimensions
	used are created.  Categories added to netCDF file.  Solar angle,
	radiation corrections.

10/01/97	2.5.2
	DAY, MONTH, YEAR added.  Fixed Production output file name, if
	FlightNum was changed.  Log File name corrected for LRT/HRT from
	setup file & if flighNum changed.

05/02/97	2.5.0
	RealTime mode added.  CAINE-II variables.

04/02/97	2.4.2
	Fixed bugs in Toggle Probe.  Tacked h onto HRT production file name.
	ASTG for PALT moved to Defaults.  Fluxes off by default.  Added
	PSURF_ALT_MAX to defaults.

11/06/96	2.4.0
	WP3 modified.  Support for new APN159.  PMSspecs file.  New high rate
	filters.  Fixed minor discrepencies in AMLIB.  Goff-Gratch for EDPC.
	Transport delays for IRS.  Dynamic lags for GPS.  Option to disable
	all lagging && despiking.

07/31/96	2.3.2
	DBZ calc fix.  Support for ex? dropped.  Production log file.
	(Supposedly) Improved file name validation.  Despike stats.  More
	B-57 AMLIB.  AKRD finalized for C-130.  Read COS blocked files.

04/14/96	2.3.1
	Differences.  New IRS parms, TKA, DRFTA, TKAR.  GIV.  B57.
	Condensed all refernce fns to one.  RATIO.  DropSonde.  Collins GPS.
	Modified AKRD for C-130.  EVTCKP renamed to EVENT_CKP.

02/11/96	2.3.0
	Fixed bug in SumCounter (was truncating), bug only affected Counters
	with calcos.  When nimbus asked for flight number, it then failed to
	record it in netCDF file.  ACE variable XFCN added. GIV.  Filled data
	gaps had time stamp of -32767:-32767:-32767, fixed.  Added 'Edit'
	Menu option to edit flight info (Date, Flt#, etc).  Added
	Dismiss/Reset to TimeSlice window.

10/23/95	2.2.1
	Added support for high-rate processing of counters sampled at 1 hz.
	Fixed bug in PSFD & INPWANT.  Flying in circles.  Allowed cal coes
	for counters.

09/15/95	2.2.0
	Fixed CheckForTimeGap to handle all time malfunctions.  260x dof alg
	change.  Added more chemistry vars for ACE.  Added spike slope to
	setup files & Edit window.  Async data file/storage added.
	Reflectivity added for FSSP-100 & ????.  Community inlet pressure
	differntial variable added.  Mod PCAS code to use FLOW for Sample
	Volume.

08/13/95	2.1.6
	Another despike bug fixed.  Checks for multiple ADS files, and
	crosses files automatically (mass store limitation).  C130 Pcor
	corrections.

06/06/95	2.1.5
	Despike bug & Sailplane bug fixed.

05/12/95	2.1.4
	Finished Sailplane stuff.

04/27/95	2.1.3
	Hard code output filename for production runs.  New code for TASHC.
	Community Inlet parameters added.

04/13/95	2.1.2
	Bug fixes, corrected winds hr bug, added fluxes to toggle probe.
	Fixed Toggle Rate bug.  X's put in front of Corrected WINDS.  TASHC.
	Sailplane stuff.  All probe's limited to 16.

??/??/95	2.1.1 - ?????

02/01/95	2.1.0
	Toggle Probe menu item. C130 PCORS.  Fixed XSO2.  New 250hz decimating
	filter (5:1 only).  Don't HR variables that don't need it.  added
	convertOffset.  New method of decodeing IRS data.

12/22/94	2.0.1
	Some bug fixes, Coeff for WP3 barrel modified and fixed all Aircraft
	WP3.

12/19/94	2.0
	Averaging modified, to apply cal coes after averaging for real-time,
	saving 10-15% in procesing of Analog channels.  Overhauled access to
	data in amlib, and modified HighRate data storage, causing 10-15%
	decrease in processing time of high rate runs.  ^R added for toggling
	output rates.  Converted to ANSI C.  Fixed bugs in batch processing.
	Despiking added.  SampleRate computation of PMS1D derived varibles.
	Fixed constants in PSURF calculation.

11/09/94	1.8
	Fixed high-rate for counter variables.  PALTF.  WIC bug fixed.
	FPCRC bug fixed.  DependedUpon, used for removing unwanted variables
	from computation.

08/31/94	1.7
	support for 1000 hz.  New PSFD code for DSM.  Disk full fix, Warning
	on non-production files.  Re-arranged some PMS1D stuff, by adding
	fake1d.c.  More C130 support.

07/15/94	1.6
	HI3 -> ALT, C130 support, base_time bug, PMS1D bin Output defaults
	TRUE, Mark Dirty vars for user to see, created file for LowRateVars.
	Bug fixes.  FLUXes to AMLIB.

04/??/94	1.5
	hr/batch bug.  New hr decimating filters.

02/??/94	1.4
	added TimeSlice time verification.  EditWindow, add full variable
	name.  More bug fixes.

02/15/94	1.3
	Removed ADS header text field.

02/10/94	1.2
	added derived variables ZERO, ONE to ALWAYS.  Straightened out the
	boys in Chemistry.

01/26/94	1.1
	released, bug fixes, changes to CVI, DPCRC, added FastOzone.

11/25/93	Nimbus 1.0 released.


nimbus/include/config.h
/*
-------------------------------------------------------------------------
OBJECT NAME:	config.h

DESCRIPTION:	Header File declaring Variable and associated processing
		functions.

NOTE:		ProcessingRate vs. HRT_Rate, there used to only be 1 HRT,
		25Hz.  I couldn't easily roll the new 50Hz into
		ProcessingRate, hence the new member HRTRate().
-------------------------------------------------------------------------
*/

#ifndef _congig_h_
#define _congig_h_

#include <string>

class Config
{
public:
  enum aircraft	
  {
	ELECTRA=308, KINGAIR=312, SABRELINER=307, C130=130, SAILPLANE=9929,
	NOAA_G4=49, HIAPER=677, NRL_P3=303, B57=357, TECHS=300, TADS=600
  };

  enum ADSVersion	{ ADS_2, ADS_3 };
  enum processingMode	{ PostProcessing, RealTime };
  enum interpolationType { Linear=0, CubicSpline, AkimaSpline };
  enum processingRate	{ SampleRate=0, LowRate=1, HighRate=25 };
  enum hrtRate		{ TwentyFive=25, Fifty=50, OneHundred=100 };
  enum groundFeedType	{ LDM, UDP };
  enum calibrationMode	{ NIDAS, NIMBUS };

  enum pms2dProcessing	{ Entire_In=0, Center_In, Reconstruction };

  Config();

  bool Interactive() const		{ return _interactive; }
  bool Despiking() const		{ return _despiking; }
  bool TimeShifting() const		{ return _timeShifting; }
  bool BlankoutVariables() const	{ return _blankout; }
  bool QCenabled() const		{ return _qcEnabled; }
  bool ProductionRun() const		{ return _productionRun; }
  bool WarnTimeLags() const		{ return _warnTimeLags; }
  bool OutputNetCDF() const		{ return _outputNetCDF; }
  bool OutputSQL() const		{ return _outputSQL; }
  bool TransmitToGround() const		{ return _transmitToGround; }
  bool EnableBroadcast() const		{ return _enableBroadcast; }
  int GroundFeedDataRate() const	{ return _transmitRate; }
  bool AsyncFileEnabled() const		{ return _asyncFileEnabled; }
  bool LoadProductionSetup() const	{ return _loadProductionSetup; }
  bool HoneyWellCleanup() const		{ return _honeywellCleanup; }
  bool InertialShift() const		{ return _inertialShift; }

  bool CreateKMLFile() const		{ return _googleEarth; }
  bool CreateNavFile() const		{ return _iwgadts; }

  bool DespikeReporting() const		{ return _despikeReporting; }
  bool LagReporting() const		{ return _lagReporting; }

  bool isADS2() const			{ return _adsVersion == ADS_2; }
  bool isADS3() const			{ return _adsVersion == ADS_3; }

  aircraft Aircraft()			{ return _aircraft; }
  std::string AircraftString();
  processingMode ProcessingMode() const	{ return _mode; }
  processingRate ProcessingRate() const	{ return _processingRate; }
  hrtRate HRTRate() const		{ return _hrtRate; }
  interpolationType InterpolationType() const { return _interpType; }
  groundFeedType GroundFeedType()	const { return _groundFeedType; }
  calibrationMode CalibrationsAppliedBy() const { return _calibrations; }

  pms2dProcessing TwoDProcessingMethod() const { return _pms2dProcessing; }
  float TwoDAreaRejectRatio() const	{ return _twoDrejectRatio; }

  size_t AnalyzeInterval() const        { return _analyzeInterval; }
  size_t VacuumInterval() const         { return _vacuumInterval; }

  void SetInteractive(bool state)	{ _interactive = state; }
  void SetProductionRun(bool state)	{ _productionRun = state; }
  void SetWarnTimeLags(bool state)	{ _warnTimeLags = state; }
  void SetDespiking(bool state)		{ _despiking = state; }
  void SetTimeShifting(bool state)	{ _timeShifting = state; }
  void SetBlankoutVariables(bool state)	{ _blankout = state; }
  void SetQCenabled(bool state)		{ _qcEnabled = state; }
  void SetOutputNetCDF(bool state)	{ _outputNetCDF = state; }
  void SetOutputSQL(bool state)		{ _outputSQL = state; }
  void SetTransmitToGround(bool state)	{ _transmitToGround = state; }
  void SetEnableBroadcast(bool state)	{ _enableBroadcast = state; }
  void SetGroundFeedDataRate(int rate)	{ _transmitRate = rate; }
  void SetAsyncFileEnabled(bool state)	{ _asyncFileEnabled = state; }
  void SetLoadProductionSetup(bool state) { _loadProductionSetup = state; }
  void SetHoneyWellCleanup(bool state)	{ _honeywellCleanup = state; }
  void SetInertialShift(bool state)	{ _inertialShift = state; }

  void SetCreateKMLFile(bool state)	{ _googleEarth = state; }
  void SetCreateNavFile(bool state)	{ _iwgadts = state; }

  void SetAircraft(aircraft ac)		{ _aircraft = ac; }
  void SetAircraft(int ac)		{ _aircraft = (aircraft)ac; }
  void SetProcessingMode(processingMode state)		{ _mode = state; }
  void SetProcessingRate(processingRate pr)		{ _processingRate = pr; }
  void SetHRTRate(hrtRate rate)				{ _hrtRate = rate; }
  void SetInterpolationType(interpolationType it)	{ _interpType = it; }
  void SetGroundFeedType(groundFeedType type)		{ _groundFeedType = type; }
  void SetCalibrationsToBeAppliedBy(calibrationMode c)	{ _calibrations = c; }

  void SetDespikeReporting(bool state)	{ _despikeReporting = state; }
  void SetLagErrorReporting(bool state)	{ _lagReporting = state; }

  void SetADSVersion(ADSVersion nv) { _adsVersion = nv; }

  void SetAnalyzeInterval(size_t ai)              { _analyzeInterval = ai; }
  void SetVacuumInterval(size_t vi)               { _vacuumInterval = vi; }

  const std::string& ProjectDirectory() const	{ return _projectDirectory; }
  const std::string& ProjectName() const	{ return _projectName; }
  const std::string& ProjectNumber() const	{ return _projectNumber; }
  const std::string& TailNumber() const		{ return _tailNumber; }
  const std::string& NIDASrevision() const	{ return _nidasRevision; }
  const std::string& FlightNumber() const	{ return _flightNumber; }
  const std::string& FlightDate() const		{ return _flightDate; }
  const std::string& ADSfileExtension() const	{ return _adsFileExtension; }
  const std::string& Checksum() const		{ return _checksum; }

  void SetProjectDirectory(const std::string s)	{ _projectDirectory = s; }
  void SetProjectName(const std::string s)	{ _projectName = s; }
  void SetProjectNumber(const std::string s)	{ _projectNumber = s; }
  void SetTailNumber(const std::string s)	{ _tailNumber = s; }
  void SetFlightNumber(const std::string s)	{ _flightNumber = s; }
  void SetFlightDate(const std::string s);
  void SetNIDASrevision(const std::string s)	{ _nidasRevision = s; }
  void SetChecksum(const std::string s)		{ _checksum = s; }

  std::string CoordinateLatitude() const        { return _coordLAT; }
  std::string CoordinateLongitude() const       { return _coordLON; }
  std::string CoordinateAltitude() const        { return _coordALT; }
  std::string CoordinateTime() const            { return _coordTime; }

  std::string WindFieldVariables() const
  { return _windWS + " " + _windWD + " " + _windWI; }

  void SetCoordLAT(const std::string s)		{ _coordLAT = s; }
  void SetCoordLON(const std::string s)		{ _coordLON = s; }
  void SetCoordALT(const std::string s)		{ _coordALT = s; }
  void SetCoordTime(const std::string s)	{ _coordTime = s; }
  void SetWindSpeed(const std::string s)	{ _windWS = s; }
  void SetWindDirection(const std::string s)	{ _windWD = s; }
  void SetWindVertical(const std::string s)	{ _windWI = s; }
  void SetTwoDProcessingMethod(pms2dProcessing p) { _pms2dProcessing = p; }
  void SetTwoDAreaRejectRatio(float f)		{ _twoDrejectRatio = f; }

  interpolationType DefaultInterpolationType();
  pms2dProcessing DefaultPMS2DProcessingMethod();
  float DefaultPMS2DAreaRatioReject();

  void SetSorterLength(int seconds) { _sorterLength = seconds; }
  int GetSorterLength() { return _sorterLength; }
  

private:
  bool _interactive;
  bool _productionRun;
  bool _warnTimeLags;
  bool _despiking;
  bool _timeShifting;
  bool _blankout;
  bool _qcEnabled;
  bool _outputNetCDF;
  bool _outputSQL;
  bool _transmitToGround;
  bool _enableBroadcast;
  bool _asyncFileEnabled;
  bool _loadProductionSetup;
  bool _honeywellCleanup;
  bool _inertialShift;

  bool _googleEarth;
  bool _iwgadts;

  bool _despikeReporting;
  bool _lagReporting;

  int _transmitRate;

  size_t _analyzeInterval;
  size_t _vacuumInterval;

  aircraft _aircraft;
  ADSVersion _adsVersion;
  processingMode _mode;
  processingRate _processingRate;
  hrtRate _hrtRate;
  interpolationType _interpType;
  groundFeedType _groundFeedType;
  calibrationMode _calibrations;	// cals to be applied by nimbus or nidas.

  std::string _projectDirectory;

  // Some meta-data.
  std::string _projectName;
  std::string _projectNumber;
  std::string _tailNumber;
  std::string _flightNumber;
  std::string _flightDate;
  std::string _nidasRevision;
  std::string _adsFileExtension;
  std::string _checksum;

  std::string _coordLAT;
  std::string _coordLON;
  std::string _coordALT;
  std::string _coordTime;
  std::string _windWS;	// Wind Speed variable
  std::string _windWD;	// Wind Direction variable
  std::string _windWI;	// Vertical Wind variable


  pms2dProcessing _pms2dProcessing;

  /* % of pixels which must be shaded inside particle bounding box.
   */
  float _twoDrejectRatio;

  // Length in seconds of the nidas processed sample sorter.
  int _sorterLength;

  static const interpolationType _defaultInterp;
  static const pms2dProcessing _defaultPMS2DProcessingMethod;
  static const float _defaultPMS2DAreaRatioReject;
};

#endif

// END CONFIG.H
